yt_dlp/extractor/common.py

```diff
@@ -3319,12 +3424,15 @@ def _merge_subtitle_items(subtitle_list1, subtitle_list2):
         return ret
 
     @classmethod
-    def _merge_subtitles(cls, subtitle_dict1, subtitle_dict2):
+    def _merge_subtitles(cls, *dicts, **kwargs):
         """ Merge two subtitle dictionaries, language by language. """
-        ret = dict(subtitle_dict1)
-        for lang in subtitle_dict2:
-            ret[lang] = cls._merge_subtitle_items(subtitle_dict1.get(lang, []), subtitle_dict2[lang])
-        return ret
+        target = (lambda target=None: target)(**kwargs)  # hack for Python 2
```

Comment:
If we didn’t have to maintain compatibility with Python 2, I’d write the signature of `_merge_subtitles` as.

```python
def _merge_subtitles(self, *dicts, target=None):
```

But Python 2 doesn’t support this syntax. So I did the next best thing: take *all* keyword arguments and filter them through this lambda to extract the one I want (and ensure there aren’t any stray ones).

Reply:
ah, gotcha. I just couldn't wrap my head around what that was doing, but now that you explain it, it is an ingenious trick.

Since this is a bit tricky to understand, maybe we should add a comment explaining this. Perhaps something like
```
Since python 2 doesnot support _merge_subtitles(self, *dicts, target=None)
we take all keyword arguments and filter them through this lambda
to extract the target keyword and ensure there aren’t any stray ones
```